articles/agora-express-token-server/index.md

```diff
@@ -0,0 +1,389 @@
+
+```JavaScript
+const expirationTimeInSeconds = 3600;
+const currentTimestamp = Math.floor(Date.now() / 1000);
+const privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds;
+```
+
+There are two types of roles in a channel, a Publisher and a Subscriber.
+
+- A Publisher is someone who streams audio/video in the channel and also receives audio/video from other publishers.
+
+- A Subscriber will not be able to stream audio/video but can receive audio/video streams from publishers.
+
+For Example, In the case of a Video Conference, everyone is a publisher. But in the case of a Livestream, only some users will be publishers while the others will be subscribers.
+
+We should get whether the user is a publisher or subscriber in the request body. The `agora-access-token` provides constants for Publisher and Subscriber.
+
+```JavaScript
+const role = req.body.isPublisher ? Agora.RtcRole.PUBLISHER : Agora.RtcRole.SUBSCRIBER;
+```
+
+We should also get the Channel ID from the request body.
+
+```JavaScript
+const channel = req.body.channel;
+```
+
+Let's pass these data to `buildTokenWithUid` to generate the token.
+
+```JavaScript
+const token = Agora.RtcTokenBuilder.buildTokenWithUid(appID, appCertificate, channel, uid, role, privilegeExpiredTs);
+```
+
+Now that we have the token, Let's return the user ID and the token as the response.
+
+```JavaScript
+res.send({ uid, token });
+```
+
+Final Code:
+
+```JavaScript
+app.post("/rtctoken", (req, res) => {
+  const appID = "<-- Your app ID here -->";
+  const appCertificate = "<-- Your app certificate here -->";
+  const expirationTimeInSeconds = 3600;
+  const uid = Math.floor(Math.random() * 100000);
+  const role = req.body.isPublisher ? Agora.RtcRole.PUBLISHER : Agora.RtcRole.SUBSCRIBER;
+  const channel = req.body.channel;
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+  const expirationTimestamp = currentTimestamp + expirationTimeInSeconds;
+
+  const token = Agora.RtcTokenBuilder.buildTokenWithUid(appID, appCertificate, channel, uid, role, expirationTimestamp);
+  res.send({ uid, token });
+});
+```
+
+Let's request this endpoint to get the RTC token now.
+
+Body of the request:
+
+```json
+{
+  "channel": "<-- Channel ID here -->",
+  "isPublisher": "<-- True or False here -->"
+}
+```
+
+Request & Response: 
+
+![RTC Token Request and Response](rtc_postman.png)
+
+> If you are using User Accounts instead of integer UIDs, you can use the `buildTokenWithAccount` function instead of the `buildTokenWithUid`.
+> In `buildTokenWithAccount`, you will pass the user account instead of the UID. Everything else remains the same. You can learn more about user accounts [here](https://docs.agora.io/en/All/faq/string).
+
+```JavaScript
+Agora.RtcTokenBuilder.buildTokenWithAccount(appID, appCertificate, channel, userAccount, role, expirationTimestamp);
+```
```

Comment:
@adrianmurage, Extremely Sorry for the inconvenience. I forgot to add this small part to the article before.
Marked it so that you wouldn't miss it.

Reply:
Accidentally started a review instead of a comment 😅